SmartDeviceLink_Example/Classes/ProxyManager.m

```diff
@@ -139,17 +133,45 @@ - (void)sdlex_showWithManager:(SDLManager *)manager {
     NSString *mainField1Text = isTextOn ? @"Smart Device Link" : @"";
     NSString *mainField2Text = isTextOn ? @"Example App" : @"";
     SDLShow* show = [[SDLShow alloc] initWithMainField1:mainField1Text mainField2:mainField2Text alignment:SDLTextAlignmentCenter];
-    show.softButtons = [self sdlex_softButtons];
-    show.graphic = areImagesVisible ? [self.class sdlex_mainGraphicImage] : [self.class sdlex_mainGraphicImage];
-    [manager sendRequest:show];
+
+    [self sdlex_softButtons:^(NSArray<SDLSoftButton *> *softButtons) {
+        show.softButtons = softButtons;
+        [manager sendRequest:show];
+    }];
+
+    [self sdlex_mainGraphicImage:^(SDLImage * _Nullable sdlImage) {
+        if (sdlImage == nil) { return ;}
+        show.graphic = sdlImage;
+        [manager sendRequest:show];
+    }];
 }
 
-- (NSArray<SDLSoftButton *> *)sdlex_softButtons {
-    SDLSoftButton *starSoftButton = [self.class sdlex_softButton1WithManager:self.sdlManager];
-    SDLSoftButton *hexagonSoftButton = [self sdlex_softButton2WithManager:self.sdlManager];
-    SDLSoftButton *textSoftButton = [self sdlex_softButton3WithManager:self.sdlManager];
-    SDLSoftButton *imageSoftButton = [self sdlex_softButton4WithManager:self.sdlManager];
-    return @[starSoftButton, hexagonSoftButton, textSoftButton, imageSoftButton];
+- (void)sdlex_softButtons:(void (^)(NSArray<SDLSoftButton *> *softButtons))completionHandler {
```

Comment:
To make this clearer, the method should be named `sdlex_prepareSoftButtons`

Reply:
Fixed